src/lib/logical_query_plan/abstract_lqp_node.cpp

```diff
@@ -126,6 +130,15 @@ void AbstractLQPNode::set_child(LQPChildSide side, const std::shared_ptr<Abstrac
   _child_changed();
 }
 
+size_t AbstractLQPNode::child_count() const {
+  /**
+   * Testing the shared_ptrs for null in _children to determine child count
+   */
+  return _children[0] ? (_children[1] ? 2 : 1) : (_children[1] ? 1 : 0);
```

Comment:
I had already suggested something here. It must have disappeared.

```c++
return _children.size() - std::count(_children.cbegin(), _children.cend(), nullptr);
```

Reply:
I like the idea, but neither looks faster nor easier to read in my opinion.

Reply:
@tbjoern @mrks  There was a discussion about this line a while ago. I proposed a different implementation but haven’t received any feedback. Is this better / worse / same?